examples/timeWarpExposureUsingTransform.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+
+#
+# LSST Data Management System
+# Copyright 2008, 2009, 2010 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+from __future__ import absolute_import, division, print_function
+from builtins import range
+import sys
+import os
+import time
+
+import lsst.utils
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+import lsst.afw.math as afwMath
+
+MaxIter = 20
+MaxTime = 1.0  # seconds
+SaveImages = False
+
+afwdataDir = lsst.utils.getPackageDir("afwdata")
+
+InputExposurePath = os.path.join(
+    afwdataDir, "ImSim/calexp/v85408556-fr/R23/S11.fits")
+
+
+def timeWarp(destMaskedImage, srcMaskedImage, srcFromDest, warpingControl):
+    """Time warpImage
+
+    int warpImage(
+        DestMaskedImageT &destMaskedImage,
+        SrcMaskedImageT const &srcMaskedImage,
+        SeparableKernel &warpingKernel, int const interpLength=0);
+
+    Parameters
+    ----------
+    destMaskedImage : `lsst.afw.image.MaskedImage`
+        Destination (output) masked image
+    srcMaskedImage : `lsst.afw.image.MaskedImage`
+        Source (input) masked image
+    srcFromDest : `lsst.afw.geom.TransformPoint2ToPoint2`
+        Transform from source pixels to destination pixels
+    warpingControl : `lsst.afw.geom.WarpingControl`
+        Warning control parameters
+
+    Returns
+    -------
+    `tuple(float, int, int)`
+        - Elapsed time in seconds
+        - Number of iterations
+        - Number of good pixels
+    """
+    startTime = time.time()
+    for nIter in range(1, MaxIter + 1):
+        goodPix = afwMath.warpImage(destMaskedImage, srcMaskedImage, srcFromDest, warpingControl)
+        endTime = time.time()
+        if endTime - startTime > MaxTime:
+            break
+
+    return (endTime - startTime, nIter, goodPix)
+
+
+def run():
+    if len(sys.argv) < 2:
+        srcExposure = afwImage.ExposureF(InputExposurePath)
+        if True:
```

Comment:
It intended to allow warping the whole image (slower) instead of a subregion. I added a global constant to this and `timeWarpExposure.py` to clarify this.